### ISSUE TITLE
Consider implementing perf counters in System.Net.Sockets

### ISSUE THREAD

**Issue by ericeil** [dotnet/corefx#7833]
The Sockets pert counter code is [currently disabled](https://github.com/dotnet/corefx/blob/master/src/System.Net.Sockets/src/System/Net/SocketPerfCounters.cs#L45).  The comments reference #2500, which covers tracing, not perf counters, and has already been closed.  We should sort out what to do about pert counters in Sockets.

**Comment by ericeil** [comment]
I guess the first question is: is there any reason to have "perf counters" for .NET Sockets in the first place?  Presumably each OS platform already offers a host of native socket monitoring/diagnostic tools that we get "for free."  

@CIPop, @davidsh, @stephentoub, @vancem, any thoughts on whether it's worth keeping custom performance counters for .NET sockets?

**Comment by ericeil** [comment]
Note that the full CLR code seems to support the following perf counters:

```csharp
    internal enum SocketPerfCounterName
    {
        SocketConnectionsEstablished,
        SocketBytesReceived,
        SocketBytesSent,
        SocketDatagramsReceived,
        SocketDatagramsSent,
    }
```

All of these look like things the underlying OS is likely to also report.

**Comment by CIPop** [comment]
> Presumably each OS platform already offers a host of native socket monitoring/diagnostic tools that we get "for free."

One thing that may be different for these is filtering the counters per process. Is there a simple/standard way to convert these to EventSource/ETW instead of removing them? Using the Windows Performance Analyzer may prove useful in certain scenarios.

**Comment by ericeil** [comment]
One obvious choice to replace perf counters is `EventCounter`.  @vancem, is this ready for production use in CoreCLR?  Is there guidance somewhere for using `EventCounter` as a perf counter replacement?

**Comment by vancem** [comment]
As Eric mentions, our expectation is to eventCoutners https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventCounter.cs found in the System.Diagnostics.Tracing package.   

This does basically what you want (you log like PerformanceCounters, but it shows up as events (which can be displayed as normal event counters).

**Comment by ericeil** [comment]
@vancem pointed out offline that an advantage to exposing these ourselves, rather than relying on the OS, is that it becomes possible to consume them from platform-agnostic code (e.g., x-plat monitoring infrastructure).  I think that's enough to justify the (relatively low) cost of this.

I'll put together a PR with an implementation.

**Comment by ericeil** [comment]
Looks like EventCounter isn't yet available in the System.Diagnostics.Tracing 4.1.0 NuGet package (#8182).

**Comment by ericeil** [comment]
Outcome of #8182: we'll need to move System.Net.Sockets forward to target NetStandard 1.5 (currently 1.3) if we want to use EventCounter.  This would mean that the implementation could only target CLR 4.6.2 or later, or .NET Core.

**Comment by davidsh** [comment]
>we'll need to move System.Net.Sockets forward to target NetStandard 1.5 (currently 1.3) if we want to use EventCounter. 

This is unrealistic right now. We need to keep System.Net.Sockets in the current generation to be compatible with UWP, etc.

**Comment by vancem** [comment]
Eric, we should talk offline about this.  I agree with David, that we don't wish to have the network stack have a dependency on NetStandard 1.5.   But a major value of having the framework broken into small DLLs is that you only need one thing upgraded you can do that.   I want to understand why that did not work in this case and see if we can fix that.

**Comment by ericeil** [comment]
I've logged issue #8223 to track making EventCounter available in NetStandard 1.3.

**Comment by ericeil** [comment]
Moving this out of this release for now; as things currently stand, it's not feasible to implement this for 1.0.  If EventCounter becomes available in NetStandard 1.3, we can reconsider.

**Comment by ericeil** [comment]
#8223 has been closed as "not possible at this time."  Closing this issue similarly.
